cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -134,13 +134,18 @@ public synchronized void create(NamespaceMeta metadata) throws Exception {
       authorizerInstantiator.get().enforce(instanceId, principal, Action.ADMIN);
     }
 
+    // store the meta first in the namespace store because namespacedlocationfactory need to look up location
+    // mapping from namespace config
```

Comment:
However, we can avoid this lookup to the store, simply because the handler (and the namespace admin doing the creating) already knows whether there's a custom mapping or not.